launcher/src/main/java/org/wildfly/core/launcher/DomainCommandBuilder.java

```diff
@@ -581,7 +581,14 @@ public DomainCommandBuilder addProcessControllerJavaOption(final String arg) {
         if (arg != null && !arg.trim().isEmpty()) {
             final Argument argument = Arguments.parse(arg);
             if (argument.getKey().equals(SECURITY_MANAGER_PROP)) {
-                setUseSecurityManager(true);
+                if (ALLOW_VALUE.equals(argument.getValue())) {
+                    // [WFCORE-5778] java.security.manager system property with value "allow" detected.
+                    // It doesn't mean SM is going to be installed but it indicates SM can be installed dynamically.
+                    setUseSecurityManager(false);
+                } else {
+                    setUseSecurityManager(true);
+                }
+                setUseSecurityManager(!ALLOW_VALUE.equals(argument.getValue()));
```

Comment:
Looks like the `if` block right above this can be removed since `setUseSecurityManager` is being set twice.

Reply:
Fixed.